lib/api/filter.rb

```diff
@@ -114,6 +128,15 @@ def parse_filter(filter)
       {:logical_or => logical_or, :operator => method, :attr => filter_attr, :value => filter_value}
     end
 
+    def composite_expression
```

Comment:
not a big fan. I liked parse only having local variables.

can you extract this logic while keeping the variables out of the global scope?
`composite_expression(and_expressions, or_expressions)`

Reply:
Humbly disagree and don't plan to change.  This is pretty idiomatic Ruby to use instance variables (not "global") like this and I think it makes definitions and callers less verbose and wordy.

For example, I picked `puma` as an example, and the first class I looked at does something pretty much like this with the first method I found:

https://github.com/puma/puma/blob/master/lib/puma/client.rb#L96-L98

In this case, the `@ready` var is called directly instead of using an `attr_reader`, but I think the general idea is the same.

* * *

So unless I get push back from others, I don't plan to change.

Reply:
lol. you've made changes like this before. it is your style.

the or expressions and and expressions is only valid within the context of `parse`
I also am not that fond of how the parsing is being performed, so elevating the local variables to globals further sets it in stone.

Can you be bribed to pass those 2 variables into the local private method `composite_expression` and do away with these ~~global~~ class variables?

Reply:
>  ~~global~~ class variables

Instance variables... they are _**instance**_ variables...

Reply:
ugh. ^ what he said